frontend/compositions/state-input/state-input.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FormEvent, useEffect, useState } from "react"
+import styles from "./state-input.module.css"
+import { FormLevelError } from '../../shared-components/index'
+import { capitalizeFirstChar } from '../../helpers/syntax-helper'
+import { states } from '../../models'
+
+interface USAStateInputProps {
+  isSubmitted: boolean
+}
+export default function USAStateInput({ isSubmitted }: USAStateInputProps) {
+  const {stateSelect, hasError} = styles
+  const [inputId, listId, errorId] = ['stateSelectInput', 'stateSelectList', 'stateSelectError']
+
+  const [inputValue, setInputValue] = useState('')
+  const [isValid, setIsValid] = useState(!isSubmitted)
+
+  function checkIsValid(value: string): boolean {
+    const matchingState = states.filter(({initials, name}) => {
+      return value.toUpperCase() === initials || capitalizeFirstChar(value) === name
```

Comment:
I'm curious about this-- it looks like we're handling the cases where a state abbrev. is submitted (i.e. MA) or a full state name is submitted (i.e. Massachusetts)-- is there a reason why we don't know which value we're getting

Reply:
Also I'm curious why we're using .filter() over .find() -- are we anticipating that there would be more than one potential result? If we used .find() we could just return !!matchingState rather than assessing an array length.

Reply:
Re: the first Q - the user can search through the provided data list either by the abbreviation or by the state name - in the event that they just type the state name/abbreviation without actually clicking to select it from the menu I don't want it to throw an error if it's otherwise valid

Re: the second Q - no reason! I guess filter just came to mind first lol - happy to switch to `.find()` to simplify the return!

Reply:
Re: the first Q - the user can search through the provided data list either by the abbreviation or by the state name - in the event that they just type the state name/abbreviation without actually clicking to select it from the menu I don't want it to throw an error if it's otherwise valid

Re: the second Q - no reason! I guess filter just came to mind first lol - happy to switch to `.find()` to simplify the return!

Reply:
Makes sense!! Thanks for clarifying! 

I do have a slight preference for .find() if you happen to be pushing again!

Reply:
Find implemented!